### PR TITLE
Chore: Upgrade typescript to version 4.3.4, storybook to 6.3.0

### PR DESCRIPTION
**Pull request by kaydelaney**
**What this PR does / why we need it**:
Upgrades typescript@4.3.4, storybook/*@6.3.0

### CODE REVIEW

File: public/app/plugins/panel/barchart/bars.ts [hunk 1]
```diff
@@ -226,7 +226,7 @@ export function getConfig(opts: BarsOptions, theme: GrafanaTheme2) {
 
       // Calculate final labels positions according to unified text size
       const textMeasurement = measureText(label.formattedValue, fontSize * devicePixelRatio);
-      const actualLineHeight = textMeasurement.fontBoundingBoxAscent + textMeasurement.fontBoundingBoxDescent;
+      const actualLineHeight = textMeasurement.actualBoundingBoxAscent + textMeasurement.actualBoundingBoxAscent;
```
Comment (leeoniya) [inline public/app/plugins/panel/barchart/bars.ts:229]: second one should be `..Descent`?

Comment (kaydelaney) [reply public/app/plugins/panel/barchart/bars.ts:229]: Ah you're right. Dangers of autocomplete 😓

**Comment by Clarity-89** [comment]
Look good and doesn't break Enterprise 🎉

**Review by Clarity-89 — APPROVED** [review]
Look good and doesn't break Enterprise 🎉

**Comment by kaydelaney** [comment]
Currently failing because of an issue with `react-docgen-typescript` that looks to be fixed here: https://github.com/styleguidist/react-docgen-typescript/pull/360

File: public/app/plugins/panel/barchart/bars.ts [hunk 2]
```diff
@@ -226,7 +226,7 @@ export function getConfig(opts: BarsOptions, theme: GrafanaTheme2) {
 
       // Calculate final labels positions according to unified text size
       const textMeasurement = measureText(label.formattedValue, fontSize * devicePixelRatio);
-      const actualLineHeight = textMeasurement.fontBoundingBoxAscent + textMeasurement.fontBoundingBoxDescent;
+      const actualLineHeight = textMeasurement.actualBoundingBoxAscent + textMeasurement.actualBoundingBoxDescent;
```
Comment (mckn) [inline public/app/plugins/panel/barchart/bars.ts:229]: nit: Is this change by accident or on purpose? Thinking that we are changing the properties being used?

**Review by mckn — COMMENTED** [review]
Great work with this @kaydelaney! Left one small question.

File: public/app/plugins/panel/barchart/bars.ts (hunk 2 above)
Comment (dprokop) [reply public/app/plugins/panel/barchart/bars.ts:229]: We can test impact of this change(if it's not accidental) in this dashboard: http://localhost:3000/d/WFlOM-jM1/barchart-panel-tests-value-sizing?orgId=1

Comment (dprokop) [reply public/app/plugins/panel/barchart/bars.ts:229]: This change seems to bring regression to bar chart value size alignment: 

| Before | After |
| --- | --- |
| image | image |

Comment (kaydelaney) [reply public/app/plugins/panel/barchart/bars.ts:229]: So the reason for this change is that 4.3.2 no longer supports this property. I couldn't find any details in typescript changelogs or blog posts as to why this was done, but I suspect it has to do with the fact that it's a non-standard property and isn't supported outside of Chrome (https://caniuse.com/?search=fontBoundingBoxAscent). I'll look into workarounds though.

**Pull request merged**